lib/services/platform-service.ts

```diff
@@ -482,8 +482,34 @@ export class PlatformService implements IPlatformService {
 	}
 
 	public deployOnEmulator(platform: string, buildConfig?: IBuildConfig): IFuture<void> {
-		this.$options.emulator = true;
-		return this.deployOnDevice(platform, buildConfig);
+		platform = platform.toLowerCase();
+
+		if (this.$options.availableDevices) {
+			return (() => {
+				let exec = require("child_process").exec;
+
+				if (this.$mobileHelper.isiOSPlatform(platform)) {
```

Comment:
This whole code looks the same.
It seems safer to check outside if this is iOS or Android and exec(command, ...) because the function after that call are same.